### PR TITLE
Issue 871 quickplot

### LINKED ISSUE #871: Improvements to `QuickPlot`
`QuickPlot` could do with a few updates. Feel free to add more to this list:

- More obvious error when two variables have different dimensions
- More obvious error when axlims are inf
- Allow plotting a single 3D variable (contour plot)
- Make the x-axis label "x" when the spatial scale isn't found (or raise error?)
- Should there be default output variables? Or should they always have to be passed in (which then makes it more obvious how to change them?)
- Try to clean up `set_output_variables` to make it more obvious what is being done
- Slider sometimes obscures the figures
- Should we be calling it `SliderPlot`?

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Improvements to QuickPlot as discussed in #871 
Might be good to check out the branch locally and try it to assess the changes

Fixes #871 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by Scottmar93** [comment]
Might be prefereable to change the axis at every time point for 2D variables. For example sim.plot(["Positive current collector potential [V]"] is just a single colour if we use the axis limits across the entire time (instead of at a specific time).

**Comment by TomTranter** [comment]
Do I need to update some widgets packages - I don't see slider on dynamic plot anymore

**Comment by tinosulzer** [comment]
@Scottmar93 ok, I will put the changing limits as default and leave the fixed ones as an option. I'll also just raise an error for 3D variables at the start

@TomTranter possibly, I have `ipywidgets==7.5.1` and `matplotlib==3.1.2`

**Comment by TomTranter** [comment]
@tinosulzer updated and still no slider in windows

File: pybamm/simulation.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import sys
 
 
-def isnotebook():
+def is_notebook():
     try:
         shell = get_ipython().__class__.__name__
         if shell == "ZMQInteractiveShell":
```
Comment (TomTranter) [inline pybamm/simulation.py]: This returns True when running in an Ipython console from spyder

**Comment by TomTranter** [comment]
@Scottmar93 @rtimms  can you quickly test this again and approve?

**Comment by tinosulzer** [comment]
I just need to figure out how to add the option for fixing axis limits or not

**Pull request merged**